src/analysis/class_builder.rs

```diff
@@ -0,0 +1,109 @@
+use crate::{
+    analysis::{
+        imports::Imports,
+        properties::{get_property_ref_modes, Property},
+        rust_type::*,
+    },
+    config::{self, GObject},
+    env::Env,
+    library,
+    traits::*,
+};
+
+pub fn analyze(
+    env: &Env,
+    props: &[library::Property],
+    type_tid: library::TypeId,
+    obj: &GObject,
+    imports: &mut Imports,
+) -> Vec<Property> {
+    let generate_builders = env
+        .config
+        .objects
+        .get(&format!("{}Builder", obj.name))
+        .is_some();
+    if !generate_builders {
+        return Vec::new();
+    }
+
+    let mut builder_properties = analyze_class(env, props, obj, imports);
+
+    for &super_tid in env.class_hierarchy.supertypes(type_tid) {
+        let type_ = env.type_(super_tid);
+
+        let super_class: &library::Class = match type_.maybe_ref() {
+            Some(super_class) => super_class,
+            None => continue,
+        };
+
+        let new_builder_properties = analyze_class(env, &super_class.properties, obj, imports);
```

Comment:
Does this still generate useless `use` statements for the parent properties?
If this is the case, it might be because we pass the wrong `obj` here.

Reply:
No its not, fixing `use`'s was main reason to separate